dht.go

```diff
@@ -288,6 +276,25 @@ func makeRoutingTable(dht *IpfsDHT, cfg config) (*kb.RoutingTable, error) {
 	return rt, err
 }
 
+// TODO This is hacky AND SHOULD be removed once https://github.com/libp2p/go-libp2p-core/pull/117
+// has landed.
+func (dht *IpfsDHT) persistRTPeersInPeerStore() {
+	tickr := time.NewTicker(peerstore.RecentlyConnectedAddrTTL / 3)
+	defer tickr.Stop()
+
+	for {
+		select {
+		case <-tickr.C:
+			ps := dht.routingTable.ListPeers()
+			for _, p := range ps {
+				dht.peerstore.AddAddrs(p, dht.peerstore.Addrs(p), peerstore.RecentlyConnectedAddrTTL)
+			}
+		case <-dht.ctx.Done():
+			return
+		}
+	}
+}
```

Comment:
I'm not convinced this should be here instead of the peerstore. What's your reasoning @aarshkshah1992 ?

In any event @Stebalien mentioned he was a little concerned about if this method will work or what issues it might cause. I think it'd be cleaner if we could just set the TTLs of peers added to the peerstore to something function of (K,alpha, refresh interval) like the `maxLastSuccessfulOutboundThreshold ` value. Apparently, the peerstore TTLs are complicated (something to definitely tackle in a future release) so we'll see how this goes.

Reply:
Hey, 

Please see https://github.com/libp2p/go-libp2p-kad-dht/pull/520#discussion_r401430829.

Also, coming up with a made-up TTL based on values based on network stats that we DO NOT have is also not a deterministic solution. What we have now is hacky but gets the job done.

Anyways, I'm going to focus on getting https://github.com/libp2p/go-libp2p/issues/800 in before 0.6 so we can get rid of this.

Reply:
I'm fine leaving this in as long as it works reasonably. The peerstore needs some quality reworking anyhow, so we can punt on the more "correct" ways of doing this for the time being.